ops/scripts/install-foundry.sh

```diff
@@ -39,11 +39,19 @@ if git rev-parse "$TAG" >/dev/null 2>&1; then
   foundryup -v "$TAG"
 else
   echo "Nightly tag doesn't exist! Building from source..."
-  foundryup -C "$SHA"
+  git checkout "$SHA"
+
+  # Use native `cargo` build to avoid any rustc environment variables `foundryup` sets. We explicitly
+  # ignore chisel, as it is not a part of `ci-builder`.
```

Comment:
> Fixes `ci-builder`'s ability to compile foundry from source when need be. Previously, we used `foundryup`, which set `RUSTFLAGS=-C target-cpu=native`. This flag fell through to the host of the container, causing invalid instruction errors. Instead, we just build with `cargo`.

For my own understanding: It seems odd that foundryup [sets](https://github.com/foundry-rs/foundry/blob/9079fb66e9483e8af6d3eff6d69e4b036d7c9691/foundryup/foundryup#L11) environment variables that make it compile incorrectly on some machines. Any idea why those use those flags?

Reply:
This is definitely what's wanted most of the time; `RUSTFLAGS=-C target-cpu=native` is great when building for your own machine. Have not dug too deep into why this flag causes the issue when building the image, but for some reason it makes it so that the target arch falls back to the host machine.

Reply:
h/t @tynes - it'd be nice for it to be `export RUSTFLAGS="${RUSTFLAGS:--C target-cpu=native}"` in `foundryup`, allowing the invoker to override this env var.

Reply:
Makes sense, https://github.com/foundry-rs/foundry/pull/7691

Reply:
ty! https://github.com/foundry-rs/foundry/pull/7692

Reply:
Lol I think we opened the same PR 😅